pydantic/networks.py

```diff
@@ -53,7 +53,7 @@ def url_regex() -> Pattern[str]:
     if _url_regex_cache is None:
         _url_regex_cache = re.compile(
             r'(?:(?P<scheme>[a-z][a-z0-9+\-.]+)://)?'  # scheme https://tools.ietf.org/html/rfc3986#appendix-A
-            r'(?:(?P<user>[^\s:/]+)(?::(?P<password>[^\s/]*))?@)?'  # user info
+            r'(?:(?P<user>[^\s:/]+)?(?::(?P<password>[^\s/]*))?@)?'  # user info
```

Comment:
```suggestion
            r'(?:(?P<user>[^\s:/]+)?'  # user
            r'(?::(?P<password>[^\s/]*))?@)?'  # password
```
as i understand it user and password are now independent, so should go on separate lines.